src/components/carousel/carousel.ts

```diff
@@ -0,0 +1,725 @@
+import { ContextProvider } from '@lit/context';
+import { LitElement, html, nothing } from 'lit';
+import {
+  property,
+  queryAll,
+  queryAssignedElements,
+  state,
+} from 'lit/decorators.js';
+
+import { type Ref, createRef, ref } from 'lit/directives/ref.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { themes } from '../../theming/theming-decorator.js';
+import IgcButtonComponent from '../button/button.js';
+import { addKeyboardFocusRing } from '../common/controllers/focus-ring.js';
+import {
+  addKeybindings,
+  arrowLeft,
+  arrowRight,
+  endKey,
+  homeKey,
+} from '../common/controllers/key-bindings.js';
+import {
+  type MutationControllerParams,
+  createMutationController,
+} from '../common/controllers/mutation-observer.js';
+import { watch } from '../common/decorators/watch.js';
+import { registerComponent } from '../common/definitions/register.js';
+import type { Constructor } from '../common/mixins/constructor.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import {
+  asNumber,
+  createCounter,
+  findElementFromEventPath,
+  isLTR,
+  partNameMap,
+} from '../common/util.js';
+import IgcIconComponent from '../icon/icon.js';
+import IgcCarouselIndicatorContainerComponent from './carousel-indicator-container.js';
+import IgcCarouselIndicatorComponent from './carousel-indicator.js';
+import IgcCarouselSlideComponent from './carousel-slide.js';
+import { carouselContext } from './context.js';
+import { styles } from './themes/carousel.base.css.js';
+import { all } from './themes/container.js';
+import { styles as shared } from './themes/shared/carousel.common.css.js';
+
+export interface IgcCarouselComponentEventMap {
+  igcSlideChanged: CustomEvent<void>;
+  igcPlaying: CustomEvent<void>;
+  igcPaused: CustomEvent<void>;
+}
+
+/**
+ * The `igc-carousel` presents a set of `igc-carousel-slide`s by sequentially displaying a subset of one or more slides.
+ *
+ * @element igc-carousel
+ *
+ * @slot Default slot for the carousel. Any projected `igc-carousel-slide` components should be projected here.
+ *
```

Comment:
Missing documentation for both **previous-button** and **next-button** slots.
Also, fill in any CSS parts if applicable.